### DISCUSSION TITLE
Clear select search input on open?

### DISCUSSION THREAD

**Discussion by wuifdesign** [mantinedev/mantine#991] [Q&A] [answered]
Is there a way to clear the search input on opening the input? Currently it will automatically filter all dropdown elements on opening a searchable select.

i want to display all options when opening it instead of only showing one and i have to delete the input to get the other options.

https://react-select.com/home does it this way.

**Comment by rtivital** [comment, marked as answer]
No, currently that is not supported

**Reply by wuifdesign** [reply]
thx for the answer. i think this is still a strange behavior, as it lets the user think he has only one option to select, but in reality it is just filtered. i dont know any other select behaving like this.

**Comment by haucongle** [comment]
Here is my fix
```javascript
<TextInput
  icon={<IconSearch />}
  placeholder="Search..."
  value={query}
  onChange={(e) => setQuery(e.currentTarget.value)}
  rightSection={
    query ? (
      <IconX
        size={18}
        style={{
          display: 'block',
          opacity: 0.5,
          cursor: 'pointer',
        }}
        onClick={() => setQuery('')}
      />
    ) : null
  }
/>
```

**Reply by pzehle** [reply]
Nice solution. Any idea on how to just show the icon without it taking styles like it is a button?

**Reply by haucongle** [reply]
> Nice solution. Any idea on how to just show the icon without it taking styles like it is a button?

`rightSection='x'`
